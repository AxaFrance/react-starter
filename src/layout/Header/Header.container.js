import React from 'react';
import { useReactOidc } from '@axa-fr/react-oidc-context';

import Header from './Header';

const EnhancedHeader = () => {
  const { oidcUser } = useReactOidc();
  const userName =
    oidcUser && oidcUser.profile
      ? oidcUser.profile.name
      : 'Utilisateur non connect√©';
  return <Header userName={userName} />;
};

export default EnhancedHeader;
