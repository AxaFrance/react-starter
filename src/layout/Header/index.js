export { HeaderApp } from './Header';
export { default } from './Header.container';
