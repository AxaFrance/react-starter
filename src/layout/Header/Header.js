import React from 'react';
import PropTypes from 'prop-types';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import { Header, Name, User } from '@axa-fr/react-toolkit-all';
import './Header.scss';

const HeaderComponent = ({ userName }) => (
  <Header>
    <Name
      title="Starter kit"
      subtitle="A little project as a demo for all our tools"
      img={logo}
      alt="Logo"
    />
    <User name={userName} href="#" profile="Inspecteur" />
  </Header>
);

HeaderComponent.propTypes = {
  userName: PropTypes.string.isRequired,
};

export default HeaderComponent;
