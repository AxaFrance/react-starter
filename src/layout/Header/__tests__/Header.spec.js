import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Header';

it('1. Renders Header component without crashing', () => {
  const { asFragment } = render(<Header userName="Jean Pascal" />);
  expect(asFragment()).toMatchSnapshot();
});
