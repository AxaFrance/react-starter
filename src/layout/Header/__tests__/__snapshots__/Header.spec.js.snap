// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Renders Header component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Starter kit
            <small
              class="af-header__subtitle"
            >
              A little project as a demo for all our tools
            </small>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <a
            class="af-info-user__link"
            href="#"
            title="Voir mon profil"
          >
            <span
              class="af-info-user__name"
            >
              Jean Pascal
            </span>
            <span
              class="af-info-user__profile"
            >
              [Inspecteur]
            </span>
          </a>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
